test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -378,6 +376,76 @@ func TestTierTemplateRevision(t *testing.T) {
 	// we add the RawExtension objects to the TemplateObjects field
+	crq := getTestCRQ("600")
+	rawTemplateObjects := []runtime.RawExtension{{Object: &crq}}
+	updateTierTemplateObjects := func(template *toolchainv1alpha1.TierTemplate) error {
+		template.Spec.TemplateObjects = rawTemplateObjects
+		return nil
+	}
+	// for simplicity, we add the CRQ to all types of templates (both cluster scope and namespace scoped),
+	// even if the CRQ is cluster scoped.
+	// WARNING: thus THIS NSTemplateTier should NOT be used to provision a user!!!
+	customTier := tiers.CreateCustomNSTemplateTier(t, hostAwait, "ttr", baseTier,
+		tiers.WithNamespaceResources(t, baseTier, updateTierTemplateObjects),
+		tiers.WithClusterResources(t, baseTier, updateTierTemplateObjects),
+		tiers.WithSpaceRoles(t, baseTier, updateTierTemplateObjects),
+		tiers.WithParameter("DEPLOYMENT_QUOTA", "60"))
+	// when
+	// we verify the counters in the status.history for 'tierUsingTierTemplateRevisions' tier
+	// and verify that TierTemplateRevision CRs were created, since all the tiertemplates now have templateObjects field populated
+	_, err = hostAwait.WaitForNSTemplateTierAndCheckTemplates(t, "ttr",
+		wait.HasStatusTierTemplateRevisions(tiers.GetTemplateRefs(t, hostAwait, "ttr").Flatten()))
+	require.NoError(t, err)
+
+	// then
+	// check the expected total number of ttr matches
+	// we IDEALLY expect one TTR per each tiertemplate to be created (clusterresource, namespace and spacerole), thus a total of 3 TTRs ideally.
+	// But since the creation of a TTR could be very quick and could trigger another reconcile of the NSTemplateTier before the status is actually updated with the reference,
+	// this might generate some copies of the TTRs. This is not a problem in production since the cleanup mechanism of TTRs will remove the extra ones but could cause some flakiness with the test,
+	// thus we assert the number of TTRs doesn't exceed the double of the expected number.
+	ttrs, err := hostAwait.WaitForTTRs(t, customTier.Name, wait.LessOrEqual(len(tiers.GetTemplateRefs(t, hostAwait, "ttr").Flatten())*2))
+	require.NoError(t, err)
+
+	t.Run("update of tiertemplate should trigger creation of new TTR", func(t *testing.T) {
+		// given
+		// that the tiertemplates and nstemlpatetier are provisioned from the parent test
+
+		// when
+		// we update one tiertemplate
+		_, err = wait.For(t, hostAwait.Awaitility, &toolchainv1alpha1.TierTemplate{}).
+			Update(customTier.Spec.ClusterResources.TemplateRef, hostAwait.Namespace, func(tiertemplate *toolchainv1alpha1.TierTemplate) {
+				// let's reduce the pod count
+				updatedCRQ := getTestCRQ("100")
+				tiertemplate.Spec.TemplateObjects = []runtime.RawExtension{{Object: &updatedCRQ}}
+			})
+		require.NoError(t, err)
+
+		// then
+		// a new TTR was created
+		updatedTTRs, err := hostAwait.WaitForTTRs(t, customTier.Name, wait.GreaterOrEqual(len(ttrs)+1))
+		require.NoError(t, err)
+
```

Comment:
could we also verify that the referenced TTR CR in the NSTemplateTier.Status.Revisions has been modified?

Reply:
makes sense, implemented in https://github.com/codeready-toolchain/toolchain-e2e/pull/1102/commits/488ef7b218d9681d4fad641fbc1d436da11602ee